app/javascript/mastodon/components/status/boost_button.tsx

```diff
@@ -24,6 +25,64 @@ import {
   selectStatusState,
 } from './boost_button_utils';
 
+// Switch between the standalone boost button or the
+// "Boost or quote" menu based on the quickBoosting preference
+export const BoostButton: FC<ReblogButtonProps> = (props) => {
+  if (quickBoosting) {
+    return <StandaloneBoostButton {...props} />;
+  }
+  return <BoostOrQuoteMenu {...props} />;
+};
```

Comment:
nit: this doesn't need to be a component, as `quickBoosting` is set on page load it can just be `BoostButton = quickBoosting ? Standalone : BoostOrQuote`